src/main/kotlin/org/openrewrite/kotlin/internal/PsiElementAssociations.kt

```diff
@@ -70,7 +70,12 @@ class PsiElementAssociations(val typeMapping: KotlinTypeMapping, val file: FirFi
     }
 
     fun primary(psiElement: PsiElement) =
-        fir(psiElement) { it.source is KtRealPsiSourceElement }
+        fir(psiElement) { filterFirElement(it) }
+
+    private fun filterFirElement(firElement : FirElement) : Boolean {
```

Comment:
Hi Kun! The error may be reproduced with:

```kotlin
val p: NotFound<String>? = null
```

compared to:
```kotlin
val p: List<String>? = null
```

I think rather than filtering out `FirUserTypeRef` we can add handling in `KotlingTypeMapping` to check if the type is resolved and return the correct type if so, otherwise return `JavaType$Unknown`.

Then the TreeParser can handle the case where the JavaType is not a JavaType$Parameterized instead of throwing an exception.

Reply:
yeah, will try to  have a separate PR for this handling

Reply:
I think we still need this change because for that real case, it's not a not-found type, but a found type.